In [1]:
import pandas as pd

In [3]:
df = pd.read_excel('Casualty.xlsx')

In [21]:
df.head()

Acc_Index  Vehicle_Reference  Casualty_Reference  Casualty_Class  \
0  197901A11AD14                  2                   1               1   
1  197901A1BAW34                  1                   1               3   
2  197901A1BFD77                  1                   1               1   
3  197901A1BFD77                  1                   2               2   
4  197901A1BFD77                  2                   3               1   

   Gender_of_Casualty  Age_Band_of_Casualty  Casualty_Severity  \
0                   1                    -1                  3   
1                   2                     6                  3   
2                   1                     5                  3   
3                   1                     4                  3   
4                   1                     6                  3   

   Pedestrian_Location  Pedestrian_Movement  Car_Passenger  \
0                    0                    0              0   
1                   10                    5              0   
2                    0                    0              0   
3                    0                    0              1   
4                    0                    0              0   

   Bus_or_Coach_Passenger  Pedestrian_Road_Maintenance_Worker  Casualty_Type  \
0                       0                                   0            104   
1                       0                                  -1              0   
2                       0                                   0            109   
3                       0                                   0            109   
4                       0                                   0            109   

   Casualty_Home_Area_Type  
0                       -1  
1                       -1  
2                       -1  
3                       -1  
4                       -1

## 1. 기록된 전체 교통사고 건수

In [5]:
len(df)

93045

## 2-1. 남성의 교통사고 건수 

In [10]:
len(df[df['Gender_of_Casualty']==1])

59778

## 2-2. 여성의 교통사고 건수

In [11]:
len(df[df['Gender_of_Casualty']==2])

32961

## 3. 사고가 일어나는 버스 안에서 가장 위험한 사람

In [12]:
df['Bus_or_Coach_Passenger'].value_counts()

 0    88722
-1     1837
 1     1371
 4      551
 3      354
 2      210
Name: Bus_or_Coach_Passenger, dtype: int64

    - 승차 중이었던 승객이 가장 위험

## 4. 전체 Casualty_Type 중 가장 많은 사고 종류는? 가장 적은 사고 종류는?

In [13]:
df['Casualty_Type'].value_counts()

109    37273
0      23006
104    14305
1       6555
11      4488
19      2866
2       2291
108      754
113      710
90       377
103      247
110      143
105       30
Name: Casualty_Type, dtype: int64

- 가장 많은 사고 종류 : 109 (차량탑승자(개인용 렌트카 포함)
- 가장 적은 사고 종류 : 105 (오토바이 - 콤비네이션 라이더 또는 조수석)

## 5. 운전자들이 가장 많이 당하는 사고 종류는? 두 번째로 많이 당하는 사고 종류는?

In [14]:
df[df['Casualty_Class']==1]['Casualty_Type'].value_counts()

109    21469
104    12802
1       6502
2       2243
19      1858
113      569
103      222
90       220
108      210
11       164
110       44
105       24
Name: Casualty_Type, dtype: int64

- 운전자들이 가장 많이 당하는 사고 종류 : 109 (차량탑승자(개인용 렌트카 포함))
- 운전자들이 두 번째로 많이 당하는 사고종류 : 104 (오토바이 라이더 또는 승객)